doc/bullet_collision_checker/src/bullet_collision_checker_tutorial.cpp

```diff
@@ -0,0 +1,315 @@
+  ros::Publisher robot_state_publisher_2(node_handle.advertise<moveit_msgs::DisplayRobotState>("robot_state_before", 1));
+  ros::Publisher world_state_publisher(node_handle.advertise<visualization_msgs::Marker>("interactive_robot_markers", 100));
+
+  {
+    // BEGIN_TUTORIAL
+    // The code starts with creating an interactive robot and a new planning scene.
+    InteractiveRobot robot;
+    g_planning_scene = new planning_scene::PlanningScene(robot.robotModel());
+
+    // Changing the collision detector to Bullet
+    // ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    // The active collision detector is set from the planning scene using the specific collision detector allocator for
+    // Bullet.
+    g_planning_scene->setActiveCollisionDetector(collision_detection::CollisionDetectorAllocatorBullet::create());
+    // For understanding the interactive robot, please refer to the Visualizing Collisions tutorial.
+    // CALL_SUB_TUTORIAL CCD
+    // CALL_SUB_TUTORIAL CCD_2
+    // END_TUTORIAL
+
+    Eigen::Isometry3d world_cube_pose;
+    double world_cube_size;
+    robot.getWorldGeometry(world_cube_pose, world_cube_size);
+    g_world_cube_shape.reset(new shapes::Box(world_cube_size, world_cube_size, world_cube_size));
+    g_planning_scene->getWorldNonConst()->addToObject("world_cube", g_world_cube_shape, world_cube_pose);
+
+    // Create a marker array publisher for publishing contact points
+    g_marker_array_publisher =
+        new ros::Publisher(node_handle.advertise<visualization_msgs::MarkerArray>("interactive_robot_marray", 100));
+
+    robot.setUserCallback(computeCollisionContactPoints);
+
+    help();
+
+    visual_tools.loadRemoteControl();
+    visual_tools.prompt("Press 'next' in the RvizVisualToolsGui window to start the continuous collision detection demo.");
+    ROS_INFO("Shutting down the interactive robot...");
+
+    // remove all collision markers
+    if (g_collision_points.markers.size())
+    {
+      for (int i = 0; i < g_collision_points.markers.size(); i++)
+        g_collision_points.markers[i].action = visualization_msgs::Marker::DELETE;
+
+      g_marker_array_publisher->publish(g_collision_points);
+    }
+
+    visualization_msgs::Marker marker_delete;
+    marker_delete.header.frame_id = "/panda_link0";
+    marker_delete.action = visualization_msgs::Marker::DELETEALL;
+    world_state_publisher.publish(marker_delete);
+
+    delete g_planning_scene;
+  }
+
+  // BEGIN_SUB_TUTORIAL CCD
+  //
+  // Continuous Collision Detection
+  // ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+  // For the CCD demonstration, the Panda robot is loaded again and with it a new planning scene created. Bullet is
+  // again set as the active collision detector.
+  robot_model::RobotModelPtr robot_model = moveit::core::loadTestingRobotModel("panda");
+  planning_scene::PlanningScenePtr planning_scene(new planning_scene::PlanningScene(robot_model));
```

Comment:
```suggestion
  auto planning_scene = std::make_shared<planning_scene::PlanningScene>(robot_model);
```

Reply:
Thanks for pointing this out, will update it. I thought that using `auto` is discouraged?

Reply:
Only when the type can't be easily inferred from context, I thought. It's clear in this case.